modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2322,6 +2325,12 @@ public String toApiName(String name) {
      */
     @Override
     public String toModelName(final String name) {
+        if (schemaKeyToModelNameCache.containsKey(name)) {
+            return schemaKeyToModelNameCache.get(name);
+        }
+
+        schemaKeyToModelNameCache.put(name, camelize(modelNamePrefix + "_" + name + "_" + modelNameSuffix));
```

Comment:
DRY. Assign ```
camelize(modelNamePrefix + "_" + name + "_" + modelNameSuffix)```
 to a variable and use it for both the add to cache and the return.

Reply:
Done